server/routes/polls/polls.route.js

```diff
@@ -138,4 +139,102 @@ router.get(
   }
 );
 
+// @route GET /polls/:pollId
+// @desc To fetch a single poll using pollId
+// @params none
+// @access private
+router.get(
+  '/:pollId',
+  passport.authenticate('jwt', { session: true }),
+  async (req, res) => {
+    const poll = await Poll.findById(req.params.pollId);
+
+    if (!poll) {
+      return res.status(404).json({ message: 'Poll not found' });
+    }
+
+    return res.status(200).json(poll);
+  }
+);
+
+// @route POST /polls/:pollId/:imageIdx/vote
+// @desc To vote on poll
+// @params none
+// @access private
+router.post(
+  '/:pollId/:imageIdx/vote',
+  passport.authenticate('jwt', { session: true }),
+  async (req, res) => {
+    const imageIdx = req.params.imageIdx;
+    if (imageIdx !== 0 || imageIdx !== 1) {
+      return res.status(401).json({
+        message: 'You must include imageIdx in API URL. And it must be 0 or 1.',
+      });
+    }
+
+    const poll = await Poll.findById(req.params.pollId);
+
+    if (!poll) {
+      return res.status(404).json({ message: 'Poll not found' });
+    }
+
+    const vote = new Vote({
+      userId: req.user._id,
+      pollId: req.params.pollId,
+      pollImageIdx: imageIdx,
+    });
+    await vote.save();
+
+    if (!vote) {
+      return res.status(400).json({ message: 'Vote failed to save' });
+    }
+
+    poll.images[imageIdx].voteIds.push(vote._id);
+    await poll.save();
+
+    return res.status(200).json(poll);
+  }
+);
+
+// @route DELETE /polls/:pollId/:imageIdx/vote
+// @desc To delete vote on poll
+// @params none
+// @access private
+router.delete(
+  '/:pollId/:imageIdx/vote',
+  passport.authenticate('jwt', { session: true }),
+  async (req, res) => {
+    const imageIdx = req.params.imageIdx;
+    if (imageIdx !== 0 || imageIdx !== 1) {
+      return res.status(401).json({
+        message: 'You must include imageIdx in API URL. And it must be 0 or 1.',
+      });
+    }
+
+    const poll = await Poll.findById(req.params.pollId);
+
+    if (!poll) {
+      return res.status(404).json({ message: 'Poll not found' });
+    }
+
+    const vote = await Vote.find({
+      $and: [{ userId: req.user._id }, { pollId: req.params.pollId }],
+    });
+
+    if (!vote) {
+      return res.status(404).json({ message: 'Failed to find vote' });
+    }
+
+    await vote.remove();
+
+    // remove voteId from image's voteIds array
+    poll.images[imageIdx].voteIds = poll.images[imageIdx].voteIds.filter(
+      (voteId) => voteId !== vote._id
+    );
```

Comment:
This looks inefficient - can you see if you could do something like this:
https://stackoverflow.com/questions/18553946/remove-sub-document-from-mongo-with-mongoose

Reply:
I did this (ref: dd48265):
```JavaScript
poll.images[imageIdx].voteIds.pull(vote._id);
await poll.save();
```
What do you think?

Reply:
Looks good